pkg/actions/actions.go

```diff
@@ -100,53 +100,51 @@ func Update(gr schema.GroupVersionResource, c *cli.Clients, opts *cli.Options) e
 }
 
 func update(gvr *schema.GroupVersionResource, dynamic dynamic.Interface, at *v1alpha1.ApprovalTask, opts *cli.Options) error {
+	// Track if user has been processed as individual User type to avoid duplicate processing
+	userProcessedAsIndividual := false
+
 	for i, approver := range at.Spec.Approvers {
 		switch v1alpha1.DefaultedApproverType(approver.Type) {
 		case "User":
 			if approver.Name == opts.Username {
-				// return true
 				at.Spec.Approvers[i].Input = opts.Input
 				if opts.Message != "" {
 					at.Spec.Approvers[i].Message = opts.Message
 				}
+				userProcessedAsIndividual = true
```

Comment:
This might still not work for cases when the group entry comes before users in the spec for the user in both user and group scenario.  I think we need to loop twice just like the change to utils.go file so that it will work for both orderings.